client/app/operations/deployments/DeploymentDetailsModalController.js

```diff
@@ -51,41 +51,59 @@ function deploymentView(deploymentRecord, clusters) {
 
     var startTime = moment(moment.utc(node.StartTime).toDate());
     var endTime = node.EndTime ? moment(moment.utc(node.EndTime).toDate()) : moment();
-
     var startEnd = startTime.format('HH:mm:ss');
     if (endTime) startEnd += ' - ' + endTime.format('HH:mm:ss');
 
     var duration = createDiffString(startTime, endTime);
     return duration + ' (' + startEnd + ')';
   }
 
-  function getViewableNodes(deployment) {
+  function getViewableNodes(deployment, expectedNodes) {
     var nodes = deployment.Value.Nodes;
-    if (nodes) {
-      return nodes.map(function (node) {
-        var logLink = '/api/v1/deployments/' + deployment.DeploymentID + '/log?account=' + deployment.AccountName + '&instance=' + node.InstanceId;
-
-        return {
-          instanceId: node.InstanceId,
-          instanceIP: node.InstanceIP,
-          status: {
-            status: node.Status,
-            lastStage: node.LastCompletedStage,
-            class: getStatusClass(node.Status)
-          },
-          duration: getNodeDuration(node),
-          numberOfAttempts: node.NumberOfAttempts || 0,
-          logLink: logLink
-        };
-      });
+    var knownNodes = nodes !== undefined ? nodes.length : 0;
+    var remaininingNodes = expectedNodes - knownNodes;
+    var viewableNodes = nodes.map(function (node) {
+      var logLink = '/api/v1/deployments/' + deployment.DeploymentID + '/log?account=' + deployment.AccountName + '&instance=' + node.InstanceId;
+      return {
+        instanceId: node.InstanceId,
+        instanceIP: node.InstanceIP,
+        status: {
+          status: node.Status,
+          lastStage: node.LastCompletedStage,
+          class: getStatusClass(node.Status)
+        },
+        duration: getNodeDuration(node),
+        numberOfAttempts: node.NumberOfAttempts || 0,
+        logLink: logLink
+      };
+    });
+
+    var failedToInitialize = deployment.Value.Status === 'Failed' || deployment.Value.Status === 'Cancelled';
+    for (var i = 0; i < remaininingNodes; i++) {
+      var n = {
+        duration: '-',
+        numberOfAttempts: '-',
+        status: {
+          status: '-', 
+        }
+      };
+      if (failedToInitialize) {
+        n.instanceId = 'Failed to initialize';
```

Comment:
Could this be confusing? It's true at the point in time at which the first failure occurred but not necessarily thereafter.